src/utils/os-resolve-group-id.js

```diff
@@ -1,8 +1,27 @@
 /** @flow */
-import { getgrnam } from 'posix';
+import uidNumber from 'uid-number';
+import logger from '../logger/logger';
+import GeneralError from '../error/general-error';
 
-export default function resolveGroupId(groupName: string): ?number {
-  const group = getgrnam(groupName);
-  if (group) return group.gid;
-  return null;
-}
+export default (async function resolveGroupId(groupName: string): Promise<?number> {
+  return new Promise((resolve, reject) => {
+    uidNumber(null, groupName, (err, uid, gid) => {
+      if (err) {
+        logger.error(err);
+        if (err.message.includes('EPERM')) {
+          return reject(
+            new GeneralError(
+              `unable to resolve group id of "${groupName}", current user does not have sufficient permissions`
+            )
+          );
+        }
+        if (err.message.includes('group id does not exist')) {
```

Comment:
@davidfirst There is no other way to do this? searching for strings in an error message is error prone.

Reply:
I know :/ I searched for an error code, but they don't provide any. It's not too bad though because all I do with it is customizing the error message. So, if I don't catch it, Bit shows the error message directly from the `uid-number` module, which is fine.